### PR TITLE
Plumb handling of Proto3Json

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This PR removes the `FieldValue.proto` property in favor of a method that accepts `useProto3Json` as an argument. Since the SDK either uses Proto3 JSON or protobufJs throughout, there is no conversion logic. 

I also updated the tests to use the new flag, which introduces two different modes. As a side effect I was able to catch a regression: The String representation of `1` and NaN didn't compare correctly.

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
Whitespace-removed diff: https://github.com/firebase/firebase-js-sdk/pull/2727/files?w=1

File: packages/firestore/src/model/field_value.ts [hunk 1]
```diff
@@ -235,6 +254,22 @@ export class DoubleValue extends NumberValue {
     }
   }
 
+  toProto(useProto3Json: boolean): api.Value {
+    if (useProto3Json) {
+      // Proto 3 let's us encode NaN and Infinity as string values as
```
Comment (dconeybe) [inline packages/firestore/src/model/field_value.ts]: Drive-by comment: "let's" should be "lets"

Comment (schmidt-sebastian) [reply packages/firestore/src/model/field_value.ts]: This was copied, so I got to apply this fixed twice. Thanks.

File: packages/firestore/src/model/field_value.ts [hunk 2]
```diff
@@ -235,6 +254,22 @@ export class DoubleValue extends NumberValue {
...
 
+  toProto(useProto3Json: boolean): api.Value {
+    if (useProto3Json) {
+      // Proto 3 lets us encode NaN and Infinity as string values as
+      // expected by the backend. This is currently not checked by our unit
+      // tests because they rely on protobuf.js.
+      if (isNaN(this.internalValue)) {
+        return { doubleValue: 'NaN' };
+      } else if (this.internalValue === Infinity) {
+        return { doubleValue: 'Infinity' } as {};
```
Comment (wilhuff) [inline packages/firestore/src/model/field_value.ts]: Why are the `as {}` required here?

File: packages/firestore/src/model/field_value.ts [hunk 3]
```diff
@@ -235,6 +254,22 @@ export class DoubleValue extends NumberValue {
     }
   }
 
+  toProto(useProto3Json: boolean): api.Value {
+    if (useProto3Json) {
+      // Proto 3 lets us encode NaN and Infinity as string values as
```
Comment (wilhuff) [inline packages/firestore/src/model/field_value.ts]: This comment reads backwards, as if it's describing the proto format, when really it's the JSON encoding of the proto that has this string form.  In other words, both encodings are proto 3 so we really need to distinguish between JSON or binary. It would be clearer if you wrote it something like this:

> In the JSON encoding, NaN and Infinity are string values ...
